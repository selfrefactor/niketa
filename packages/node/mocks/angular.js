export const angularMock = {
  stdout: '----------|---------|----------|---------|---------|-------------------\n' +
    'File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s \n' +
    '----------|---------|----------|---------|---------|-------------------\n' +
    'All files |     100 |      100 |     100 |     100 |                   \n' +
    ' foo.ts   |     100 |      100 |     100 |     100 |                   \n' +
    '----------|---------|----------|---------|---------|-------------------\n',
  stderr: "ts-jest[versions] (WARN) Version 25.1.0 of jest installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=24.0.0 <25.0.0). Please do not report issues in ts-jest if you are using unsupported versions.\n" +
    'PASS src/app/helpers/foo.spec.ts\n' +
    '  âœ“ happy (1ms)\n' +
    '\n' +
    'Test Suites: 1 passed, 1 total\n' +
    'Tests:       1 passed, 1 total\n' +
    'Snapshots:   0 total\n' +
    'Time:        1.465s\n' +
    'Ran all test suites matching /\\/home\\/s\\/repos\\/joke-maker-angular\\/src\\/app\\/helpers\\/foo.spec.ts/i.\n'
}

export const angularWithLog ={}
export const angularError ={}
export const angularErrorWithLog ={}