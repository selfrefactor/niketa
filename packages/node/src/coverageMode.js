import { allTrue, getter, ok , take } from 'rambdax'

import { clean } from './_modules/clean'
import { parseCoverage } from './_modules/parseCoverage'
import { shouldNotify } from './_modules/shouldNotify'
import { takeNotifyWhenError } from './ants/takeNotifyWhenError'

import { additional } from './emitters/additional'
import { show } from './emitters/show'
import { tooltip } from './emitters/tooltip'

const ERROR_ICON = '‚ùå'
const SUCCESS_ICON = 'üê∏'
const ERROR_CONDITION = 'LINE === undefined'

// Run coverage and send to `niketa-notify` and VSCode
// ============================================
export function coverageMode({
  emit,
  execResult,
  fileName,
  filePath,
  maybeSpecFile,
  notify,
  notifyClose,
}){
  if(execResult.stderr.startsWith('FAIL') || execResult.stderr.includes('ERROR:')){
    const notifyWhenError = takeNotifyWhenError(execResult)
    if(allTrue(notifyWhenError,notify, notifyClose)){
      notify(notifyWhenError)
      notifyClose()
    }
    tooltip(emit, take(800,execResult.stderr))
    return show(emit, ERROR_ICON)
  }

  const { pass, message, uncovered } = parseCoverage(
    execResult,
    fileName,
    filePath
  )
  ok(message)('string')
  if (message === ERROR_CONDITION) return show(emit, SUCCESS_ICON)

  show(emit, pass ? message : ERROR_ICON)
  const cleaner = clean(execResult, pass, uncovered)
  
  if (cleaner.stdout.trim() === '') return

  const okNotify = allTrue(
    shouldNotify(maybeSpecFile),
    cleaner.stdout.includes('console.log'),
    getter('electron.connected')
  )

  if (okNotify){
    notify(cleaner.stdout)
    notifyClose()
  }
  
  tooltip(emit, `${ cleaner.stderr }${ cleaner.stdout }${ cleaner.uncovered }`)
  additional(emit, uncovered)
}