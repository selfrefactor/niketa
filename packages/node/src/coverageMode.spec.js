import { angularMock } from '../mocks/angular.js'
import { coverageMode } from './coverageMode'

const filePathSpec =
  '/home/s/repos/joke-maker-angular/src/app/helpers/foo.spec.ts'
const filePath=   '/home/s/repos/joke-maker-angular/src/app/helpers/foo.spec.ts'

test('happy', () => {
  const emit = jest.fn()
  const notify = jest.fn()
  const notifyClose = jest.fn()
  const fileName = 'foo'

  const result = coverageMode({
    emit,
    execResult    : angularMock,
    fileName,
    filePath: filePathSpec,
    maybeSpecFile : filePathSpec,
    notify,
    notifyClose,
  })
  expect(1).toBe(1)
})
