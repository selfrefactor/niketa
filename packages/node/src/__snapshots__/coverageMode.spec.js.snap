// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`happy 1`] = `
Array [
  Array [
    Object {
      "channel": "show",
      "message": "🐰 st: 100 br: 100 fn: 100 lns: 100",
    },
  ],
  Array [
    Object {
      "channel": "tooltip",
      "message": "ts-jest[versions] (WARN) Version 25.1.0 of jest installed has not been tested with ts-jest. If you're experiencing issues, consider using a supported version (>=24.0.0 <25.0.0). Please do not report issues in ts-jest if you are using unsupported versions.
----------|---------|----------|---------|---------|-------------------
File      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------|---------|----------|---------|---------|-------------------
All files |     100 |      100 |     100 |     100 |                   
 foo.ts   |     100 |      100 |     100 |     100 |                   
----------|---------|----------|---------|---------|-------------------
",
    },
  ],
  Array [
    Object {
      "channel": "additional",
      "message": "",
    },
  ],
]
`;
